src/libwalletqt/Wallet.cpp

```diff
@@ -699,6 +739,26 @@ Q_INVOKABLE QString Wallet::getSpendProof(const QString &txid, const QString &me
     return QString::fromStdString(result);
 }
 
+void Wallet::getSpendProofAsync(const QString &txid, const QString &message, const QJSValue &ref) const
+{
+  QFuture<QString> future = QtConcurrent::run(this, &Wallet::getSpendProof, txid, message);
+  auto watcher = new QFutureWatcher<QString>();
```

Comment:
```suggestion
  auto watcher = new QFutureWatcher<QString>(this);
```

Reply:
QFutureWatcher takes non-const QObject as a parent. In order to do this I had to make getSpendProofAsync non-const